canvas_modules/common-canvas/src/object-model/redux/reducers/palette.js

```diff
@@ -28,7 +28,7 @@ export default (state = {}, action) => {
 	case "REMOVE_NODE_TYPES_FROM_PALETTE":
 	case "SET_CATEGORY_LOADING_TEXT":
 	case "SET_CATEGORY_EMPTY_TEXT": {
-		return Object.assign({}, state, { categories: categories(state.categories, action) });
+		return Object.assign({}, state, { categories: categories(state.categories, action), version: "3.0" });
```

Comment:
Why are you saving the version here?

Reply:
Typically, when a palette is loaded into common-canvas it is has the `version: "3.0"` property set or, if it is an older version, common-canvas upgrades it to v3.0 and then loads it. v1.0 of the palette schema did not have a version property defined so the absence of that property should indicate a v1.0 palette.

When using the 'save to palette' option on a node I noticed that, if there is no current palette loaded, the action would create a palette in Redux correctly with a new category and the node in it, but the in-memory palette would not have a version property.  

This means that, if the palette was saved to a file and then later loaded into common-canvas, the absence of a version number would make common-canvas think that the palette was a v1.0 palette (because we didn't have the version property for v1.0). Common-canvas would then have problems loading that palette because the absence of the version property should indicate the palette is v1.0 but the actual contents (category and node) comply with v3.0. Consequently, I added the version property in the line above to avoid that problem.